File: system-test/dotnet/SystemTests/Legacy/Mod06/SmokeTests/System/MyShopSmokeTest.cs
using Common;
using SystemTests.Legacy.Mod06.Base;
using Dsl.Core.UseCase;
using Optivem.Testing;
using Shouldly;
using Xunit;

namespace SystemTests.Legacy.Mod06.SmokeTests.System;

public class MyShopSmokeTest : BaseChannelDriverTest
{
    [Theory]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    public async Task ShouldBeAbleToGoToMyShop(Channel channel)
    {
        await SetChannelAsync(channel);

        var result = await _shopDriver!.GoToMyShopAsync();
        result.ShouldBeSuccess();
    }
}

File: system-test/java/src/test/java/com/mycompany/myshop/systemtest/legacy/mod06/smoke/system/MyShopSmokeTest.java
package com.mycompany.myshop.systemtest.legacy.mod06.smoke.system;

import com.mycompany.myshop.systemtest.legacy.mod06.base.BaseChannelDriverTest;
import com.mycompany.myshop.testkit.channel.ChannelType;
import com.mycompany.myshop.testkit.driver.port.dtos.GoToMyShopRequest;
import com.optivem.testing.Channel;
import org.junit.jupiter.api.TestTemplate;

import static com.mycompany.myshop.testkit.common.ResultAssert.assertThatResult;

class MyShopSmokeTest extends BaseChannelDriverTest {
    @TestTemplate
    @Channel({ChannelType.UI, ChannelType.API})
    void shouldBeAbleToGoToMyShop() {
        var result = myShopDriver.goToMyShop(GoToMyShopRequest.builder().build());
        assertThatResult(result).isSuccess();
    }
}

